fsm/README.md

```diff
@@ -0,0 +1,198 @@
+# go-statemachine FSM Module
+
+A well defined DSL for constructing finite state machines for use in Filecoin.
+
+
+## Table of Contents
+
+* [Background Reading](./README.md#background-reading)
+* [Description](./README.md#description)
+* [Usage](./README.md#usage)
+* [Interaction With Statestore](./README.md#interaction-with-statestore)
+* [License](./README.md#license)
+
+## Background Reading
+
+[Wikipedia - Finite State Machines](https://en.wikipedia.org/wiki/Finite-state_machine)
+[Wikipedia - UML State Machine](https://en.wikipedia.org/wiki/UML_state_machine)
+
+## Description
+
+This library provides a way to model a Filecoin process that operates on a specific data structure as a series of state transitions. It is used in the storage and retrieval markets in [go-fil-markets repository](https://github.com/filecoin-project/go-fil-markets) to model the lifecycle of a storage or retrieval deal. It may be used for other parts of Filecoin in the future.
+
+A state machine is defined in terms of
+
+- StateType -- the type of data structure we are tracking (should be a struct)
```

Comment:
Note: Some things in this list, e.g. Events, are backtick'ed - and others (like StateKeyField) are not.